buffer/src/main/java/io/netty/buffer/ByteBufUtil.java

```diff
@@ -588,7 +588,7 @@ static ByteBuf encodeString0(ByteBufAllocator alloc, boolean enforceHeap, CharBu
             dst = alloc.buffer(length);
         }
         try {
-            final ByteBuffer dstBuf = dst.internalNioBuffer(0, length);
+            final ByteBuffer dstBuf = dst.internalNioBuffer(dst.readerIndex(), length);
```

Comment:
I think if the buffer not has a reader index of 0 its a bug as we ask for a new buffer. That said the change is ok as it makes it more consistent with other usages